src/libopensc/card-cac.c

```diff
@@ -1469,6 +1476,48 @@ static int cac_populate_cac_1(sc_card_t *card, int index, cac_private_data_t *pr
 	return SC_SUCCESS;
 }
 
+static int cac_process_ACA(sc_card_t *card, cac_private_data_t *priv)
+{
+	int r, index;
+	u8 *tl = NULL, *val = NULL;
+	size_t tl_len, val_len;
+
+
+	r = cac_read_file(card, CAC_FILE_TAG, &tl, &tl_len);
+	if (r < 0)
+		goto done;
+
+	r = cac_read_file(card, CAC_FILE_VALUE, &val, &val_len);
+	if (r < 0)
+		goto done;
+
+	/* TODO we should process the ACA file -- so far we are happy we can read it */
+	//r = cac_parse_ACA(card, priv, tl, tl_len, val, val_len);
+	r = cac_find_first_pki_applet(card, &index);
+        if (r == SC_SUCCESS) {
+		priv = cac_new_private_data();
+		if (!priv) {
+			r = SC_ERROR_OUT_OF_MEMORY;
+			goto done;
+		}
+		r = cac_populate_cac_1(card, index, priv);
+		if (r == SC_SUCCESS) {
+			priv->aca_path = malloc(sizeof(sc_path_t));
```

Comment:
are you leaking memory here by not freeing `priv->aca_path`?

Reply:
The `aca_path` is already freed in `cac_free_private_data()` from last time. This commit is mostly refactoring the code from `cac_find_and_initialize()` to separate function.

Reply:
OK, just wanted to be sure